### PR TITLE
✨ Update analytics vendor split experiments config

### PR DESCRIPTION
**Pull request by jonathantyng-amp**
- Add `gulp generate-vendor-configs` to experiments command so that vendor JSONs are generated during the experiment build
- Enable `--compile_vendor_configs` by default, so no need to add this flag anymore

### CODE REVIEW

**Comment by jonathantyng-amp** [comment]
cc @zhouyx for review :)

cc @erwinmombay just want to make sure, is running 2 commands during experiment builds okay? e.g. `gulp generate-vendor-jsons && gulp dist --defineExperimentConstant=ANALYTICS_VENDOR_SPLIT`

**Comment by zhouyx** [comment]
A few notes on this PR:

1. Our release script always run `gulp generate-vendor-jsons` and `gulp dist --compile_vendor_configs`. 
By adding `gulp generate-vendor-jsons` to the experiment build command, we will be generating vendor jsons twice. But that should be fine given it only take a few seconds. 
Future idea is that we want to remove the `gulp generate-vendor-jsons` task from everywhere after launching the experiment. 

2. Remove `compile_vendor_configs` in this PR. Because the flag itself is easy to be missed. Given in the future all build will need to copy the json file to the `dist` folder. we decide to remove the flag and make it a default process during build. 

3. The reason we are making the change now is because we want to unblock @estherkim's work on integration test. We figure it's easier to modify the experiment build command instead of rewriting the default build command in travis.

File: build-system/tasks/extension-helpers.js [hunk 1]
```diff
@@ -386,7 +386,7 @@ function buildExtension(
   }
 
   // minify and copy vendor configs for amp-analytics component
-  if (name === 'amp-analytics' && argv.compile_vendor_configs) {
+  if (name === 'amp-analytics') {
```
Comment (zhouyx) [inline build-system/tasks/extension-helpers.js]: @jonathantyng-amp could you please confirm that the build works even if `gulp generate-vendor-jsons` hasn't been run (no such json files to copy). In that case what will happen? Will the analytics-vendors folder be created? Thanks.

Comment (jonathantyng-amp) [reply build-system/tasks/extension-helpers.js]: Tested locally and confirmed that build works with no JSON files to copy. If there are no vendor JSONs, `dist/analytics-vendors` folder will not be created

Comment (zhouyx) [reply build-system/tasks/extension-helpers.js]: Great! Thanks

**Pull request merged**